src/Cache/_ScsDataClient.php

```diff
@@ -354,14 +369,92 @@ public function listErase(string $cacheName, string $listName, ?int $beginIndex=
         }
         return new CacheListEraseResponseSuccess();
     }
+
+    public function dictionarySet(string $cacheName, string $dictionaryName, string $field, string $value, bool $refreshTtl, ?int $ttlSeconds = null): CacheDictionarySetResponse
+    {
+        try {
+            validateCacheName($cacheName);
+            validateDictionaryName($dictionaryName);
+            validateFieldName($field);
+            validateValueName($value);
+            $ttlMillis = $this->ttlToMillis($ttlSeconds);
+            $dictionarySetRequest = new _DictionarySetRequest();
+            $dictionarySetRequest->setDictionaryName($dictionaryName);
+            $dictionarySetRequest->setItems([$this->toSingletonFieldValuePair($field, $value)]);
+            $dictionarySetRequest->setRefreshTtl($refreshTtl);
+            $dictionarySetRequest->setTtlMilliseconds($ttlMillis);
+            $call = $this->grpcManager->client->DictionarySet($dictionarySetRequest, ["cache" => [$cacheName]], ["timeout" => $this->deadline_seconds * self::$TIMEOUT_MULTIPLIER]);
+            $this->processCall($call);
+        } catch (SdkError $e) {
+            return new CacheDictionarySetResponseError($e);
+        } catch (Exception $e) {
+            return new CacheDictionarySetResponseError(new UnknownError($e->getMessage()));
+        }
+        return new CacheDictionarySetResponseSuccess();
+    }
+
+    private function toSingletonFieldValuePair(string $field, string $value): _DictionaryFieldValuePair
+    {
+        $singletonPair = new _DictionaryFieldValuePair();
+        $singletonPair->setField($field);
+        $singletonPair->setValue($value);
+        return $singletonPair;
+    }
+
+    public function dictionaryGet(string $cacheName, string $dictionaryName, string $field): CacheDictionaryGetResponse
+    {
+        try {
+            validateCacheName($cacheName);
+            validateDictionaryName($dictionaryName);
+            validateFieldName($field);
+            $dictionaryGetRequest = new _DictionaryGetRequest();
+            $dictionaryGetRequest->setDictionaryName($dictionaryName);
+            $dictionaryGetRequest->setFields([$field]);
+            $call = $this->grpcManager->client->DictionaryGet($dictionaryGetRequest, ["cache" => [$cacheName]], ["timeout" => $this->deadline_seconds * self::$TIMEOUT_MULTIPLIER]);
+            $dictionaryGetResponse = $this->processCall($call);
+        } catch (SdkError $e) {
+            return new CacheDictionaryGetResponseError($e);
+        } catch (Exception $e) {
+            return new CacheDictionaryGetResponseError(new UnknownError($e->getMessage()));
+        }
+        if ($dictionaryGetResponse->hasMissing()) {
+            return new CacheDictionaryGetResponseMiss();
+        }
+        if ($dictionaryGetResponse->getFound()->getItems()->count() == 0) {
+            return new CacheDictionaryGetResponseError(new Exception("_DictionaryGetResponseResponse contained no data but was found"));
```

Comment:
The exception used to construct the Error response class needs to be an SdkError like the one above this on line 428. Please use `UnknownError` like you did above unless you think one of the other SdkError-derived classes works better.